src/wasm-linker.cpp

```diff
@@ -366,10 +369,30 @@ bool hasI64ResultOrParam(FunctionType* ft) {
   return false;
 }
 
+void Linker::makeDummyFunction() {
+  bool create = false;
+  // Check if there are indirectly called functions
+  for (auto& relocation : out.relocations) {
+    if (relocation->kind == LinkerObject::Relocation::kFunction) {
+      create = true;
+      break;
+    }
+  }
+  if (!create) return;
+  wasm::Builder wasmBuilder(out.wasm);
+  Expression *unreachable = wasmBuilder.makeUnreachable();
+  Function *dummy = wasmBuilder.makeFunction(Name(dummyFunction), std::vector<NameType>(), WasmType::none, std::vector<NameType>(), unreachable);
+  functionIndexes[dummy->name] = 0;
+  out.wasm.table.names.push_back(dummy->name);
```

Comment:
instead of adding to the table manually, you can do `ensureFunctionIndex`. perhaps it should be moved from an `auto` in a method to a proper method on the object.